cmd/pulumi-test-language/tests/testdata/l2-resource-keyword-overlap/main.pp

```diff
@@ -0,0 +1,56 @@
+resource "class" "simple:index:Resource" {
+  value = true
+}
+
+output "class" {
+  value = class
+}
+
+resource "export" "simple:index:Resource" {
+  value = true
+}
+
+output "export" {
+  value = export
+}
+
+resource "mod" "simple:index:Resource" {
+  value = true
+}
+
+output "mod" {
+  value = mod
+}
+
+resource "import" "simple:index:Resource" {
+  value = true
+}
+
+# TODO(pulumi/pulumi#18246): Pcl should support scoping based on resource type just like HCL does in TF so we can uncomment this.
+# output "import" {
+#   value = Resource["import"]
```

Comment:
I don't follow, why do we want to use this instead of `value = import` here?

I was somehow thinking that `mod` was a keyword in Go, but that's only the case in `go.mod`, not in `*.go` files it seems. Unless we have the `import` case working, we have nothing that ensures we "escape" invalid identifiers in Go.

Maybe we could add a test case that uses `if` as a name? That should hit all languages 🤔

Reply:
Sorry to be nitpicking here, but could we add a case here using `if`, or another Go keyword that does not clash with HCL/PCL, so that we can validate the renaming code works in Go?